### ISSUE TITLE
Sequence id unwrapping is flawed, and may result in wrong extrapolated sequence numbers

### ISSUE THREAD

**Issue by wdouglass** [carnegierobotics/LibMultiSense#43]
consider the following function
```
const int64_t& impl::unwrapSequenceId(uint16_t wireId)
{
    //
    // Look for a sequence change

    if (wireId != m_lastRxSeqId) {

        const uint16_t ID_MAX    = std::numeric_limits<uint16_t>::max();
        const uint16_t ID_CENTER = ID_MAX / 2;

        //
        // Seed

        if (-1 == m_lastRxSeqId)
			m_unWrappedRxSeqId = m_lastRxSeqId = wireId;

        //
        // Detect forward 16-bit wrap

        else if (wireId        < ID_CENTER   &&
                 m_lastRxSeqId > ID_CENTER) {

            m_unWrappedRxSeqId += 1 + (ID_MAX - m_lastRxSeqId) + wireId;

        //
        // Normal case

        } else
            m_unWrappedRxSeqId += wireId - m_lastRxSeqId;

        //
        // Remember change

        m_lastRxSeqId = wireId;
    }

    return m_unWrappedRxSeqId;
}
```

if `wireId` is `ID_MAX / 2 - 1` and `m_lastRxSeqId` is `ID_MAX / 2 + 1`, intuitively it looks like packets were reordered.

This code will assume a rollover, and chaos will ensue

i propose the following change:
we replace `ID_CENTER = ID_MAX / 2` with `ID_LOW = ID_MAX / 8` and `ID_HIGH = ID_MAX - ID_LOW`
and then replace ` (wireId < ID_CENTER   &&  m_lastRxSeqId > ID_CENTER)` with ` (wireId < ID_LOW   &&  m_lastRxSeqId > ID_HIGH)`

This would increase the reordering tolerance considerably

An alternate solution is to retool libmultisense to expect 16 bit sequence numbers instead of 64 bit and deal with reordering in some other way (possibly by adding a timeout to the udp tracker cache). this solution is probably more robust (because there aren't any stateful assumptions in packet interpretation), but would take more resources to design and test.

Let me know what you think

**Comment by wdouglass** [comment]
it looks like the udp cache only carries 10 messages, so just letting 16 bits roll over is probably fine...

**Comment by QuentinTorg** [comment]
Closed by #45
